models/stg_iterable__user_unsub_message_type.sql

```diff
@@ -18,22 +18,34 @@ fields as (
         */
         {{
             fivetran_utils.fill_staging_columns(
-                source_columns=adapter.get_columns_in_relation(ref('stg_iterable__user_unsub_message_type_history_tmp')),
-                staging_columns=get_user_unsubscribed_message_type_history_columns()
+                source_columns=adapter.get_columns_in_relation(ref('stg_iterable__user_unsub_message_type_tmp')),
+                staging_columns=get_user_unsubscribed_message_type_columns()
             )
         }}
         
     from base
 ),
 
 final as (
-    
+
     select 
-        email,
-        message_type_id,
-        updated_at,
+
+        cast(_fivetran_id as {{ dbt.type_string() }} ) as _fivetran_user_id,
+        coalesce(_fivetran_id, email) as unique_user_key,
+        cast(message_type_id as {{ dbt.type_string() }} ) as message_type_id,
+        {{ dbt_utils.generate_surrogate_key(['_fivetran_id', 'email', 'message_type_id','updated_at']) }} as unsub_message_type_unique_key,
+        
+        {% if does_table_exist('user_unsubscribed_message_type') == false %}
+        rank() over(partition by email order by updated_at desc) as latest_batch_index,
+        {% else %}
+        1 as latest_batch_index,
+        {% endif %}
```

Comment:
Resorting to this because when trying to do the coalesce method in the final CTE, we run into a redshift error in both the partition by and the order by when it's a null field. The coalesce doesn't work for the order_by because the 2 fields need to be the same data type

 No matching signature for function COALESCE for argument types: TIMESTAMP, STRING.

And there's not another timestamp field.